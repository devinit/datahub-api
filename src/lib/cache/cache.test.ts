import 'jest';
import {readCacheData, ICached} from '.';
import * as prettyFormat from 'pretty-format';
import * as path from 'path';

describe('cache module tests', () => {
    const cachePath = path.resolve(__dirname, './test-data/.cache');

    it('should read precache file and return an array of JS objects', async () => {
        try {
            const cachedData: ICached[] = await readCacheData(cachePath);
            expect(cachedData[0].key).toBe('global-picture/themes.csv');
            expect(prettyFormat(cachedData)).toMatchSnapshot();
        } catch (error) {
            console.log(error);
        }
    });
});
