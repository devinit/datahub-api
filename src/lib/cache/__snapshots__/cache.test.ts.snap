// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cache module tests should read precache file and return an array of JS objects 1`] = `
"Array [
  Object {
    \\"key\\": \\"global-picture/themes.csv\\",
    \\"type\\": \\"cms\\",
  },
  Object {
    \\"key\\": \\"global/entity.csv\\",
    \\"type\\": \\"cms\\",
  },
]"
`;
